airflow/providers/mysql/hooks/mysql.py

```diff
@@ -52,7 +52,10 @@ def __init__(self, *args, **kwargs) -> None:
 
     def set_autocommit(self, conn: Connection, autocommit: bool) -> None:  # noqa: D403
```

Comment:
in the code here, presently this `Connection` refers to the airflow model connection but this is not correct

it should be this one: `from MySQLdb.connections import Connection`

now might be a good opportunity to fix this, and also make it a `Union` with the mysql-connector connection type also

Reply:
Do you have to import both modules before you can use their names for a `Union` typing? If so, I would need to figure out if having them both imported at the same time would cause any issues. It looks like the `get_conn()` method is only importing one module depending on the value of `client_name`

Reply:
again this is a good point.

the `mysql` extra installs both connector python and mysqldb so it should be safe to import both.  i don't think we save much by importing locally.

i'll ask around though.  please write a test for this also -- it is testable behavior and i think merits it

Reply:
Generally: one of the biggest slow downs in Airflow "boot" time is importing modules, so avoiding top level imports where it's not strictly needed is better I think.

Reply:
Ok, given that, I guess we should just remove that type annotation

Reply:
Ash provided another option in slack ....

You could put the annotation import behind `if TYPING:`

I think either way is ok

Reply:
btw i explored this idea...

this seems to at least run (and it accomlishes the goal of providing correct type annotation without importing...

not sure how mypy will feel about it but you could try it 

and not sure if it is exactly what ash is suggesting.

```python
TYPING = False
if TYPING:
    from MySQLdb.connections import Connection as MySQLdbConnection
    from mysql.connector.abstracts import MySQLConnectionAbstract
...
    def set_autocommit(
        self, conn: Union['MySQLdbConnection', 'MySQLConnectionAbstract'], autocommit: bool
    ) -> None:
```

Reply:
Added in recent commits, but not sure how to actually test it.

Reply:
ok i can try to look later today

Reply:
@ericpp i know you were working through the right way to test this autocommit fix, but it seems you added a bunch of tests that are unrelated to this change?  can you remove the tests that don't have anything to do with this PR? perhaps i'm missing something?

after you do that, i'll take a look and see if we need to make any adjustments, or if you have specific questions i can try to address them.

Reply:
The `TestMySqlHookMySqlConnectorPython` class is a version of the existing `TestMySqlHook` class that works for the `mysql-connector-python` client. It seems like the existing `TestMySqlHook` class only supports the `MySQLdb` client. 

I moved the code that works for both clients from `TestMySqlHook` into `TestMySqlBulk`

I did add a `test_get_autocommit` method to both test classes and can remove that if needed.

Reply:
oh i see thanks lemme look

Reply:
> I moved the code that works for both clients from TestMySqlHook into TestMySqlBulk

ok maybe not 'moved' but 'copied'

Reply:
The test_bulk methods were moved into the `TestMySqlBulk` class. The git diff looks strange because it plopped my changes into middle of the old `TestMySql` class.

Reply:
I just removed the `test_get_autocommit` test that I did add to the code.